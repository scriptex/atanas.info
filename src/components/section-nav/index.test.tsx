import * as React from 'react';
import { render } from '@testing-library/react';

import { SectionNav } from '.';

describe('SectionNav component', () => {
	it('Should render the SectionNav component', () => {
		const { asFragment, container } = render(
			<SectionNav
				name="title"
				data={[
					{
						url: 'https://google.com',
						title: 'Title 1',
						description: 'Description 1'
					},
					{
						url: 'https://google.com',
						title: 'Title 2',
						description: 'Description 2'
					}
				]}
				active={0}
				onClick={jest.fn()}
			/>
		);

		expect(asFragment()).toMatchSnapshot();

		container.querySelector<HTMLElement>('.c-btn')?.click();

		expect(asFragment()).toMatchSnapshot();
	});
});
