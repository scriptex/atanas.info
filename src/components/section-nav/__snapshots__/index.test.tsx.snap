// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionNav component Should render the SectionNav component 1`] = `
<DocumentFragment>
  <nav
    class="c-section__nav"
  >
    <ul>
      <li
        class="current"
      >
        <button
          aria-label="Switch to Title 1"
          class="c-btn"
          type="button"
        >
          Title 1
        </button>
      </li>
      <li>
        <button
          aria-label="Switch to Title 2"
          class="c-btn"
          type="button"
        >
          Title 2
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`SectionNav component Should render the SectionNav component 2`] = `
<DocumentFragment>
  <nav
    class="c-section__nav"
  >
    <ul>
      <li
        class="current"
      >
        <button
          aria-label="Switch to Title 1"
          class="c-btn"
          type="button"
        >
          Title 1
        </button>
      </li>
      <li>
        <button
          aria-label="Switch to Title 2"
          class="c-btn"
          type="button"
        >
          Title 2
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
